web/src/app/mod.rs

```diff
@@ -265,22 +265,22 @@ mod test {
         let mut req = Request::default();
         req.extensions_mut().insert(Foo);
 
-        let res = service.call(req).await.unwrap();
+        let res = Service::<Request<_>>::call(&service, req).await.unwrap();
 
         assert_eq!(res.status().as_u16(), 200);
         assert_eq!(res.headers().get(CONTENT_TYPE).unwrap(), TEXT_UTF8);
 
         let mut req = Request::default();
         *req.uri_mut() = Uri::from_static("/abc");
 
-        let res = service.call(req).await.unwrap();
+        let res = Service::<Request<_>>::call(&service, req).await.unwrap();
```

Comment:
It seems that type inference have a hard time with such highly generic code. Though i wouldn't mind some turbofish api to guide inference for `Req` like with `GenericRouter::with_custom_object`.

But why it doesn't require annotations in `Context` tests. Needs more investigation and experimenting with other possible variations.